asserts/database.go

```diff
@@ -176,3 +223,110 @@ func (db *Database) Check(assert Assertion) error {
 	}
 	return fmt.Errorf("failed signature verification: %v", lastErr)
 }
+
+func (db *Database) checkAssertionRevisionAndClearLatest(indexPath string, newRev int) error {
+	curRevStr, err := db.readlinkEntry(assertionsRoot, indexPath, "latest")
+	if err != nil {
+		return fmt.Errorf("broken assertion storage, reading 'latest' revision symlink: %v", err)
+	}
+	curRev, err := strconv.Atoi(curRevStr)
+	if err != nil {
+		return fmt.Errorf("broken assertion storage, extracting revision from 'latest' revision symlink: %v", err)
+	}
+	if curRev >= newRev {
+		return ErrNotSuperseding
+	}
+	err = db.removeEntry(assertionsRoot, indexPath, "latest")
```

Comment:
This seems broken, though. We're not dealing with an atomic database, so we cannot afford to lose knowledge like this. If the system is reset while we are in the interim state, it will all break down.